test/extensions/filters/listener/common/fuzz/listener_filter_fuzzer.cc

```diff
@@ -58,18 +62,23 @@ void ListenerFilterFuzzer::fuzz(
         .WillByDefault(testing::InvokeWithoutArgs([&got_continue]() { got_continue = true; }));
 
     while (!got_continue) {
-      if (header.done()) { // End of stream reached but not done
-        file_event_callback_(Event::FileReadyType::Closed);
+      if (data.done()) { // End of stream reached but not done
+        if (events_ & Event::FileReadyType::Closed) {
+          file_event_callback_(Event::FileReadyType::Closed);
+        }
         return;
       } else {
         file_event_callback_(Event::FileReadyType::Read);
       }
+
+      data.next();
     }
   }
 }
 
-FuzzedHeader::FuzzedHeader(const test::extensions::filters::listener::FilterFuzzTestCase& input)
-    : nreads_(input.data_size()), nread_(0) {
+FuzzedInputStream::FuzzedInputStream(
+    const test::extensions::filters::listener::FilterFuzzTestCase& input)
+    : nreads_(input.data_size()) {
```

Comment:
This can be done in a follow-up PR if you want. One good way to make sure that all your constructors initialize all members correctly is to have one that does all the actual work and make the others do some transformation of the input and then call that one. So here, you could call the new constructor like this:
```
FuzzedInputStream::FuzzedInputStream(const test::extensions::filters::listener::FilterFuzzTestCase& input)
    : FuzzedInputStream(extractBuffer(input), extractIndices(input)) {}
```

where `extractBuffer` and `extractIndices` are private static functions that basically do what is currently done in this method body.

Reply:
Yup, let's do this in a follow-up PR.

Reply:
+1